### DISCUSSION TITLE
Welcome to cvui Discussions!

### DISCUSSION THREAD

**Discussion by Dovyski** [Dovyski/cvui#105] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by coldstart01** [comment]
Hi Dovyski,

The cvui framework seems to live up to its promises. I have one question today though. If I want to draw on a frame via a function, what is my frame type, and could this be done:
main.cpp:
`cvui::printf(frame, 1215, 80, 0.4, 0xffffff, "K = [%.3Lf, %.3Lf, %.3Lf", CV_FX, 0.0, CV_CX);
`
could I call it:
`print_it(frame)
`
in function section:
`
print_it(frame) ??but what type?
{
CV_FX = 0;
CV_FY = 0;
cvui::printf(frame, 1215, 80, 0.4, 0xffffff, "K = [%.3Lf, %.3Lf, %.3Lf", CV_FX, 0.0, CV_CX);
}
`

thx for your time

**Reply by Dovyski** [reply]
Hi! Thank you for your interest on cvui. Regarding your question, if I understood it correctly, you could create a function like:

```cpp
void print_it(cv::Mat frame) { ... }
```

The `frame` should be an OpenCV Matrix (`cv::Mat`). Since this function is printing something into that frame, there is no need to return a value (so `void` as a return should be ok).

**Comment by ulva1011** [comment]
Hi Dovyski,
i have found cvui and it seems like the best GUI-Tool for my case for C++ with OpenCV.
I want to learn a bit more with the examples but i do not get them built.
Well i am not really a programmer and have not much clue of GIT Bash/CMake.
When typing the directions/instruction like (git clone....... , mkdir build.......) it seems to work.
But after "cmake .. -DCMAKE....................." it says "Configuring incomplete, errors occured!" and when typing "make -j4" it repeats "bash: make: command not found.
Can you help me to get it work. :/

Here is the GIT Bash Textfile
It says anything with an OpenCVConfig-cmak File. Well but i am not used to this stuff enough to know much about.
[GIT-Errors.txt](https://github.com/Dovyski/cvui/files/8755228/GIT-Errors.txt)
